### PR TITLE
ENG 1765 Unify table appearance

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
- CSV dialog now uses PaginatedTable.
- Improve CSV parsing to handle delimiters in the data itself.
- No longer need to add an id column, so removed it.
- Change inferSchema to take in a defaultType parameter so that the default type can be specified (results in quotes around everything and it looks weird for the CSV dialog).

## Related issue number (if any)
ENG 1765

## Loom demo (if any)
Before:
Notice the last row. The full row is: 4,16,"Didn't see the wall, oops!" but it is cut off due to the comma delimiter in the comment.

Screen Shot 2023-01-06 at 9 48 27 AM

After:
Notice the last row. We can now see the full message: Didn't see the wall, oops!

Screen Shot 2023-01-06 at 9 48 05 AM

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/DataPreviewer/index.tsx [hunk 1]
```diff
@@ -1,169 +0,0 @@
-import Alert from '@mui/material/Alert';
```
Comment (agiron123) [inline src/ui/common/src/components/DataPreviewer/index.tsx:1]: I take it this file is no longer used?

Comment (eunice-chan) [reply src/ui/common/src/components/DataPreviewer/index.tsx:1]: Yes, I did CRTL + F DataPreviewer and I didn't see it used anywhere.

Comment (agiron123) [reply src/ui/common/src/components/DataPreviewer/index.tsx:1]: Sounds good 👌

**Comment by eunice-chan** [comment]
@agiron123 
1. Created a Linear task for this.
2. Added padding to the bottom. See image below.
Screen Shot 2023-01-10 at 11 38 47 AM

File: src/ui/common/src/components/integrations/dialogs/csvDialog.tsx [hunk 2]
```diff
@@ -24,51 +25,63 @@ export const CSVDialog: React.FC<Props> = ({ setDialogConfig, setErrMsg }) => {
...
-      if (headerName === 'id') {
-        hideColumn = true;
+    let regexResp;
+    // for each regexp match (either comma, newline, or quoted item)
+    while ((regexResp = splitFinder.exec(text))) {
+      const split = regexResp[0];
+
+      // if it's not a quote capture, add an item to the current row
+      // (quote captures will be pushed by the newline or comma following)
+      if (split.startsWith(`"`) === false) {
```
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/csvDialog.tsx:53]: nit: can do !split.startsWith(`"`)

**Review by agiron123 — APPROVED** [review]

**Pull request merged**